Classes/Middleware/LanguageRedirectMiddleware.php

```diff
@@ -122,4 +134,39 @@ protected function findMatchingPresetByAcceptLanguageHeader(string $acceptLangua
 
         return null;
     }
+
+    /**
+     * Match frontend language cookie against given language dimensions.
+     * Return the Locale, that fits best.
+     *
+     * @param string $acceptLanguageHeader
+     *
+     * @return array|null
+     */
+    protected function findMatchingPresetByFeLanguageCookie(string $feLanguageCookie): ?array
+    {
+        $detectedLocale = $this->localeDetector->detectLocaleFromLocaleTag($feLanguageCookie);
+
+        if (!$detectedLocale instanceof Locale) {
```

Comment:
Could you please move L150-168 to a new function and use it in both findMatchingPresetBy... methods?
Sth. like:
```php
$detectedLocale = 'dependsOnMethod';

$preset = $this->findPresetByLocale($detectedLocale);

// For the presetByLanguageCookie one can directly return $preset.
// This check is only neccessary in the AcceptLanguageHeader while loop.
if ($preset !== null) {
    return $preset;
}

// ...
```